src/pymatgen/analysis/adsorption.py

```diff
@@ -300,7 +302,7 @@ def find_adsorption_sites(
             sites = [site + distance * np.asarray(self.mvec) for site in sites]
 
             ads_sites[key] = sites
-        ads_sites["all"] = sum(ads_sites.values(), [])  # noqa: RUF017
+        ads_sites["all"] = functools.reduce(operator.iadd, ads_sites.values(), [])
```

Comment:
generally not so sure this ruff rule is actually an improvement. more verbose and less readable imo. maybe just ignore `RUF017` in general?

Reply:
I think this ruff is implemented to reduce time complexity ([`Checks for the use of sum() to flatten lists of lists, which has quadratic complexity.`](https://docs.astral.sh/ruff/rules/quadratic-list-summation/)), happy to revert this if you believe readability is more important than performance for these cases.

Reply:
good to know. didn't realize `sum` scales differently. i would say let's go with `[item for sublist in lists for item in sublist]` then since it doesn't require extra imports and list comprehension is often the most performant in python

Reply:
Wonderful idea. It looks like [list comprehension performs pretty well](https://realpython.com/python-flatten-list/#considering-performance-while-flattening-your-lists):

```
Time to flatten a 1000x1000 matrix (in milliseconds):

flatten_concatenation()..........1.95 ms
flatten_extend().................2.03 ms
flatten_reduce_iconcat().........2.68 ms
flatten_chain()..................4.60 ms
flatten_comprehension()..........7.79 ms  >> with list comprehension 
flatten_sum().................1113.22 ms
flatten_reduce_concat().......1117.15 ms
flatten_reduce_lambda().......1117.52 ms
flatten_reduce_add()..........1118.80 ms
```